hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/RegionFactoryDefaultTest.java

```diff
@@ -46,6 +61,200 @@ protected Properties getCacheProperties() {
     }
 
     @Test
+    public void testEntity() {
+        final HazelcastInstance hz = getHazelcastInstance(sf);
+        assertNotNull(hz);
+        final int count = 100;
+        final int childCount = 3;
+        insertDummyEntities(count, childCount);
+        sleep(1);
+        List<DummyEntity> list = new ArrayList<DummyEntity>(count);
+        Session session = sf.openSession();
+        try {
+            for (int i = 0; i < count; i++) {
+                DummyEntity e = (DummyEntity) session.get(DummyEntity.class, (long) i);
+                session.evict(e);
+                list.add(e);
+            }
+        } finally {
+            session.close();
+        }
+        session = sf.openSession();
+        Transaction tx = session.beginTransaction();
+        try {
+            for (DummyEntity dummy : list) {
+                dummy.setDate(new Date());
+                session.update(dummy);
+            }
+            tx.commit();
+        } catch (Exception e) {
+            tx.rollback();
+            e.printStackTrace();
+        } finally {
+            session.close();
+        }
+
+        Statistics stats = sf.getStatistics();
+        Map<?, ?> cache = hz.getMap(DummyEntity.class.getName());
+        Map<?, ?> propCache = hz.getMap(DummyProperty.class.getName());
+        Map<?, ?> propCollCache = hz.getMap(DummyEntity.class.getName() + ".properties");
+        assertEquals((childCount + 1) * count, stats.getEntityInsertCount());
+        // twice put of entity and properties (on load and update) and once put of collection
+        // TODO: fix next assertion ->
+        //        assertEquals((childCount + 1) * count * 2, stats.getSecondLevelCachePutCount());
+        assertEquals(childCount * count, stats.getEntityLoadCount());
+        assertEquals(count, stats.getSecondLevelCacheHitCount());
+        // collection cache miss
+        assertEquals(count, stats.getSecondLevelCacheMissCount());
+        assertEquals(count, cache.size());
+        assertEquals(count * childCount, propCache.size());
+        assertEquals(count, propCollCache.size());
+        sf.getCache().evictEntityRegion(DummyEntity.class);
+        sf.getCache().evictEntityRegion(DummyProperty.class);
+        assertEquals(0, cache.size());
+        assertEquals(0, propCache.size());
+        stats.logSummary();
+    }
+
+    @Test
+    public void testQuery() {
```

Comment:
I agree with you, some tests look too long. However the logic they implement are really brief. Just insert some entities, read to have them cached, manipulate them a little and read again to check if they are retrieved from cache or not. Anyway, the long tests you mentioned were not written by me. I just moved them. If they need to get shortened this can be done in another pr after I get this one merged.